docs/AVS-Guide.md

```diff
@@ -0,0 +1,85 @@
+- The AVS provider now is aware of the change in stake, and the staker is free to complete their withdrawal.
+
+The following figure illustrates the above flow: 
+![Stake update](./images/staker_withdrawing.png)
+
+### *Deregistering from AVS*
+In order for any EigenLayer operator to be able to de-register from a AVS, EigenLayer provides the interface `recordLastStakeUpdateAndRevokeSlashingAbility(..)`. Essentially, in order for an operator to deregister from a AVS, the operator has to call `recordLastStakeUpdateAndRevokeSlashingAbility(..)` in EigenLayer's `Slasher.sol` via the AVS's ServiceManager contract. It is important to note that the latest block number until which the operator is required to serve tasks for the service must be known by the service and included in the ServiceManager's call to `Slasher.recordLastStakeUpdateAndRevokeSlashingAbility`.
+
+The following figure illustrates the above flow in which the operator calls the `deregister(..)` function in a sample Registry contract.
+![Operator deregistering](./images/operator_deregister.png)
+
+### *Slashing*
+As mentioned above, EigenLayer is built to support slashing as a result of an on-chain-checkable, objectively attributable action. In order for a AVS to be able to slash an operator in an objective manner, the AVS needs to deploy a DisputeResolution contract which anyone can call to raise a challenge against an EigenLayer operator for its adversarial action. On successful challenge, the DisputeResolution contract calls the `ServiceManager.freezeOperator(..)`; the ServiceManager in turn calls `Slasher.freezeOperator(..)` to freeze the operator in EigenLayer. EigenLayer's Slasher contract emits a `OperatorFrozen(..)` event whenever an operator is (successfully) frozen
+
+The following figure illustrates the above flow: 
+![Slashing](./images/slashing.png)
+
+
+## Quick Start Guide to Build AVS Contracts:
+The EigenLayer team has built a set of reusable and extensible contracts for use in AVSs built on top of EigenLayer. These are contained in the general-purpose [/middleware/ folder](https://github.com/Layr-Labs/eigenlayer-contracts/tree/master/src/contracts/AVS), which contains code that can be extended, used directly, or consulted as a reference in building AVS on top of EigenLayer. There are several basic contracts that all AVS-specific contracts can be built on:
+- The *VoteWeigherBase contract* tracks an operator’s “weight” in a given quorum, across all strategies that are associated with that quorum.  This contract also manages which strategies are in each quorum - this includes both removing and adding functionalities as well as changing strategy weights.  
```

Comment:
confusing wording: "both removing and adding functionalities as well as changing strategy weights" => "functionalities for both adding and removing strategies, as well as changing strategy weights"

To me the existing wording implies that the contract manages adding and removing new **functionalities themselves**, rather than adding and **strategies in quorums**